conans/test/util/tools_test.py

```diff
@@ -576,6 +576,49 @@ def vcvars_constrained_test(self):
                 # Not raising
                 tools.vcvars_command(settings, force=True)
 
+    def vcvars_context_manager_test(self):
+        conanfile = """
+from conans import ConanFile, tools
+
+class MyConan(ConanFile):
+    name = "MyConan"
+    version = "0.1"
+    settings = "os", "compiler"
+
+    def build(self):
+        with tools.vcvars(self.settings):
+            self.output.info(str(self.settings.os) + ": vcvars applies")
+            self.output.info("VCINSTALLDIR set to: " + str(tools.get_env("VCINSTALLDIR")))
+"""
+        profile_win = """
```

Comment:
Probably you don't need to create profiles (specially if the check is not based on settings), just use the default of the testing machine and it should work anyway.